src/Orchard.Cms.Web/Program.cs

```diff
@@ -1,20 +1,50 @@
 using System.IO;
+using System.Linq;
+using System.Reflection;
 using System.Threading;
+using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Modules;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
 using Orchard.Hosting;
+using Orchard.Logging;
 
 namespace Orchard.Cms.Web
 {
     public class Program
     {
         public static void Main(string[] args)
         {
-            var host = new WebHostBuilder()
-                .UseIISIntegration()
-                .UseKestrel()
-                .UseContentRoot(Directory.GetCurrentDirectory())
+            var host = WebHost.CreateDefaultBuilder(args)
+                .ConfigureAppConfiguration((hostingContext, config) =>
+                {
+                    var env = hostingContext.HostingEnvironment;
+                    config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile("logging.json", optional: true, reloadOnChange: true)
+                          .AddJsonFile($"logging.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+
+                    if (env.IsDevelopment())
+                    {
+                        var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
```

Comment:
This piece I copied from `CreateDefaultBuilder` implementation.
By the way, maybe it is better to copy all and do not call `CreateDefaultBuilder`, I don't know